### DISCUSSION TITLE
I need to restart the script watcher too often

### DISCUSSION THREAD

**Discussion by ramiroaraujo** [johnlindquist/kit#1344] [Errors] [answered]
I'm not sure why, I haven't debugged it at all, but I find myself either deleting the .js file (I use .ts) and forcing scriptkit to rebuild it or I restart the script watcher. 
How can I help debug this?

Thanks!

**Comment by ramiroaraujo** [comment]
Actually, the only way right now is to delete the .mjs file in .scripts. Otherwise it doesn't update it

**Reply by johnlindquist** [reply]
Which version number are you on? The latest version doesn't use the script watcher anymore

**Reply by ramiroaraujo** [reply]
1.79.5

**Reply by johnlindquist** [reply]
I guess I hadn't published it publicly yet 😅

Here's the latest: https://github.com/johnlindquist/kitapp/releases/tag/v1.94.14

Trying to get everything polished up and release it as Script Kit v2

**Reply by ramiroaraujo** [reply]
I'll check it out ASAP and let you know :)

**Reply by ramiroaraujo** [reply]
Yey! this new version seems to have fixed it, for the moment. I'll try it out a bit more during today, and mark it as answered if all ok.
BTW, there's still the "Debug -> Restart Script Watcher" option in the menu, not sure if it makes sense anymore

**Reply by johnlindquist** [reply, marked as answer]
@ramiroaraujo 

> BTW, there's still the "Debug -> Restart Script Watcher" option in the menu, not sure if it makes sense anymore

Nice, thanks!

Yeah, this version doesn't use file watching at all anymore. It uses a custom loader (kind of like ts-node) every time you launch a script built with TypeScript. So no .mjs files, no typescript file-watching, just a custom loader.

Edit: Clarification -> We still use "file watching" for reading script metadata, updating the scripts.json, etc, but not for the typescript compilation.

**Reply by ramiroaraujo** [reply]
BTW @johnlindquist this newer version works _much_ better. The debugger is very reliable now (though I'd love to integrate it with my IDE with regular breakpoints instead of `debugger` 🙈), and the overall experience and I think performance are better.
Looking forward to the new version!!
